flink-runtime/src/main/java/org/apache/flink/runtime/metrics/dump/MetricDumpSerialization.java

```diff
@@ -50,12 +51,27 @@
 	private MetricDumpSerialization() {
 	}
 
+	public static class MetricSerializationResult {
```

Comment:
Can you add a short class level comment about why we added this? The problem with determining number of metrics before hand etc.